### DISCUSSION TITLE
CRSFforArduino and Teensy

### DISCUSSION THREAD

**Discussion by hw999** [ZZ-Cat/CRSFforArduino#50] [General]
Hi,
I tried to compile for a Teensy 3.2 and noticed that the latest release 0.4.0 does not include Teensies as compatible hardware. 
Could you please give a hint what I can do to solve the problem ? Would a change of the compatibilty table solve the problem ? 
I could also use a Teensy 4.0 if the 3.2 is a problem.

**Comment by hw999** [comment]
Hi
disregard my previous post. I got somewhat lost in the documentation and thought that 0.4.0 was the latest version. Version 0.5.0 has the Teensy in it and the channel-example works. Looking forward to work on the telemetry part.

**Reply by ZZ-Cat** [reply]
Full telemetry support was also added in 0.5.0.  
Currently, I am going through the process of writing up the documentation for the API via Docusaurus (bit of a learning curve, for me).

**Comment by hw999** [comment]
A big thanks for sharing your library. Did some initial testing on the telemetry part and it works just fine. And programming is far more sophisticated than the stuff I produce.

Is CRSF-telemetery a bidirectional system - two wires from receiver to host (in my case the Teensy) required ? Looks like - pulling one of the wires make it stop.

The Crossfire/Tracer-system provides a couple of outputs. I configured a Crossfire-receiver on two pads for CRSF output for telemetry only and on another pad for SBUS output which is used as input for a redundancy box. Is something similar also possible under ELRS ? I don't know much about this system but the little I know tells me that it is not possible because a ELRS-rx is not meant to have more than one output and mixing of protocols is not implemented. Is this correct ?

**Reply by ZZ-Cat** [reply]
You're very welcome. =^/,..,^=

CRSF, as I know it, is bidirectional.  
The reason why it seems to stop when you pull one of the wires is because telemetry is sent out directly after an RC Channels packet has been received. This is how Betaflight, RotorFlight and nearly every other flight controller firmware (that uses CRSF) does it.

If there is a way to spoof "Telemetry Only", I may consider introducing an option to configure my library for it. However, that would be heavily dependent on whether-or-not the CRSF Protocol will allow it.  
I have seen another attempt at this. But, with that implementation, the code still relied on receiving RC Channels (and discarding them, instead of parsing them) to trigger sending telemetry packets.

As for doing redundancy/daisy-chaining ExpressLRS recievers (if that's what I think you're referring to), I _have_ heard of it being done. Although, I have not yet done this myself, so I cannot confidently answer your question here. You may have better luck asking the ExpressLRS developers.  

As for "mixing protocols", it can't be done, at this time.
